analysis/filter/MergeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/mergefilter/LargeMerge.kt

```diff
@@ -0,0 +1,54 @@
+package de.maibornwolff.codecharta.filter.mergefilter
+
+import de.maibornwolff.codecharta.model.BlacklistItem
+import de.maibornwolff.codecharta.model.Edge
+import de.maibornwolff.codecharta.model.Node
+import de.maibornwolff.codecharta.model.NodeType
+import de.maibornwolff.codecharta.model.Project
+
+class LargeMerge {
+    companion object {
+        fun packageProjectInto(project: Project, prefix: String): Project {
+            val modifiedProject = Project(
+                project.projectName,
+                moveNodesIntoFolder(project.rootNode, prefix),
+                project.apiVersion,
+                addFolderToEdgePaths(project.edges, prefix),
+                project.attributeTypes,
+                project.attributeDescriptors,
+                addFolderToBlackListPaths(project.blacklist, prefix)
+            )
+            return modifiedProject
+        }
+
+        private fun moveNodesIntoFolder(root: Node, folderName: String): List<Node> {
+            val newRoot = Node(
+                name = folderName,
+                type = NodeType.Folder,
+                children = root.children
+            )
+            val mutableRoot = root.toMutableNode()
+            mutableRoot.children = mutableSetOf(newRoot.toMutableNode())
+            return listOf(mutableRoot.toNode())
+        }
+
+        private fun addFolderToEdgePaths(edges: List<Edge>, folderName: String): List<Edge> {
+            edges.forEach {
+                it.fromNodeName = insertFolderIntoPath(it.fromNodeName, folderName)
+                it.toNodeName = insertFolderIntoPath(it.toNodeName, folderName)
+            }
+            return edges
+        }
+
+        private fun addFolderToBlackListPaths(blacklist: List<BlacklistItem>, folderName: String): List<BlacklistItem> {
+            blacklist.forEach {
+                it.path = insertFolderIntoPath(it.path, folderName)
+            }
+            return blacklist
+        }
+
+        private fun insertFolderIntoPath(path: String, folderName: String): String {
+            return path.replace(Regex("^/root/"), "/root/$folderName/")
```

Comment:
This assumes that the path starts with /root/. Do you know if we should check that? If one file is corrupted, it might produce an error that affects everything.

Reply:
Christian wasn't sure about it, and I couldn't find a project where it is different. 
At first I had a `require` there with a `/root/` check, but then I needed a test case for something that might not exist in reality. 
So the question is:
- Dynamically insert after the first "folder" e.g. root
- require root

What do you think?

Reply:
I am fine with either. But better be prepared than sorry. In all cases, it's a simple check that prevents it from exploding.